x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/get_generated_title.ts

```diff
@@ -72,20 +74,25 @@ export function getGeneratedTitle({
     }).pipe(
       hide(),
       concatenateChatCompletionChunks(),
+      last(),
       map((concatenatedMessage) => {
-        const input =
+        let title: string =
           (concatenatedMessage.message.function_call.name
             ? JSON.parse(concatenatedMessage.message.function_call.arguments).title
             : concatenatedMessage.message?.content) || '';
 
-        // This regular expression captures a string enclosed in single or double quotes.
+        // This captures a string enclosed in single or double quotes.
         // It extracts the string content without the quotes.
         // Example matches:
         // - "Hello, World!" => Captures: Hello, World!
         // - 'Another Example' => Captures: Another Example
         // - JustTextWithoutQuotes => Captures: JustTextWithoutQuotes
-        const match = input.match(/^["']?([^"']+)["']?$/);
-        const title = match ? match[1] : input;
+        if (
+          (title.startsWith(`"`) && title.endsWith(`"`)) ||
+          (title.startsWith(`'`) && title.endsWith(`'`))
+        ) {
+          title = title.substring(1, title.length - 1);
+        }
```

Comment:
This ought to still be accomplishable by a regex. The simple approach which mostly works:

```js
title = title.replace(/^["'](.*)["']$/g, '$1');
```

If we don't want to rewrite strings with mixed quotes (`"foo'`) we need something like:

```
title = title.replace(/^"(.*)"$/g, '$1').replace(/^'(.*)'$/g, '$1');
```

That being said, sometimes an if is simpler

Reply:
This one failed previously: https://github.com/elastic/kibana/pull/182923/files/d653ab8bcc15e51dcb44149621dfa5a25b3c046c#diff-1a334051ae31ea647d550d3bd0bcddf84f4338c5b296f19bb6b04a7075afeda8R115

I'll take your code, I think. It removes an if() branch.